nifi-nar-bundles/nifi-extension-utils/nifi-hadoop-utils/src/main/java/org/apache/nifi/processors/hadoop/AbstractHadoopProcessor.java

```diff
@@ -563,8 +558,8 @@ protected UserGroupInformation getUserGroupInformation() {
     /*
      * Overridable by subclasses in the same package, mainly intended for testing purposes to allow verification without having to set environment variables.
      */
-    String getAllowExplicitKeytabEnvironmentVariable() {
-        return System.getenv(ALLOW_EXPLICIT_KEYTAB);
+    boolean isAllowExplicitKeytab() {
+        return Boolean.parseBoolean(System.getenv(ALLOW_EXPLICIT_KEYTAB));
```

Comment:
Does this throw an exception if its null or something besides true/false?

Wondering if it should be `"true".equals(System.getenv(ALLOW_EXPLICIT_KEYTAB))`

Reply:
Boolean.parseBoolean(String s) doesn't throw an exception.  It'll return false unless the string passed to it, ignoring case, equals "true".  It doesn't do any trimming, just takes the given string and evaluates it:
```java
    public static boolean parseBoolean(String s) {
        return ((s != null) && s.equalsIgnoreCase("true"));
    }
```